Add Cookstyle 6.3 notes (#610)
chef/cookstyle: RELEASE_NOTES.md

## Cookstyle 6.3

### 2 New Cops

#### ChefModernize/RespondToCompileTime

> The `ChefModernize/RespondToCompileTime` detects gating usage of the `compile_time` property in the `chef_gem` resource with `if respond_to?(:compile_time)`. The compile_time property was added in Chef Infra Client 12.1 and can safely be used without checking first.

Reviewer: ```suggestion
The `ChefModernize/RespondToCompileTime` cop detects gating usage of the `compile_time` property in the `chef_gem` resource with `if respond_to?(:compile_time)`. The `compile_time` property was added in Chef Infra Client 12.1 and can safely be used without checking first.
```

Author: Fixed

> The `ChefModernize/RespondToCompileTime` detects gating usage of the `compile_time` property in the `chef_gem` resource with `if respond_to?(:compile_time)`. The compile_time property was added in Chef Infra Client 12.1 and can safely be used without checking first.

Reviewer 2: ```suggestion
The `ChefModernize/RespondToCompileTime` cop detects gating usage of the `compile_time` property in the `chef_gem` resource with `if respond_to?(:compile_time)`. The compile_time property was added in Chef Infra Client 12.1 and can safely be used without checking first.
```

`Enabled by default`: True

`Autocorrects`: Yes

#### ChefDeprecations/DeprecatedShelloutMethods

> The `ChefDeprecations/DeprecatedShelloutMethods` cop detects the deprecated shell_out helper methods that were removed in Chef Infra Client 15. These helpers were not widely used and were never publically documented, but they are still occasionally found in cookbooks.

Reviewer 2: ```suggestion
The `ChefDeprecations/DeprecatedShelloutMethods` cop detects deprecated `shell_out` helper methods that were removed in Chef Infra Client 15. These helpers were not widely used and were never publicly documented, but they are still occasionally found in cookbooks.
```

`Enabled by default`: True

`Autocorrects`: No

### RuboCop 0.82

> The RuboCop engine that powers Cookstyle has been updated from 0.81.0 to 0.82.0 which includes several important bug fixes that impacted cookbooks.

Reviewer: ```suggestion
The RuboCop engine that powers Cookstyle has been updated from 0.81.0 to 0.82.0. This update includes several important bug fixes that impacted cookbooks.
```
